backend/src/v5/processors/teamspaces/projects/models/commons/tickets.groups.js

```diff
@@ -14,22 +14,39 @@
  *  You should have received a copy of the GNU Affero General Public License
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+/* eslint-disable no-underscore-dangle */
 
 const { UUIDToString, stringToUUID } = require('../../../../../utils/helper/uuids');
-const { getGroupById, updateGroup } = require('../../../../../models/tickets.groups');
+const { addGroups, getGroupById, updateGroup } = require('../../../../../models/tickets.groups');
+const { getMetadataByQuery, getMetadataByRules } = require('../../../../../models/metadata');
+const { getNodesByIds, getNodesBySharedIds } = require('../../../../../models/scenes');
 const { getFile } = require('../../../../../services/filesManager');
 const { getLatestRevision } = require('../../../../../models/revisions');
-const { getMetadataByRules } = require('../../../../../models/metadata');
-const { getNodesBySharedIds } = require('../../../../../models/scenes');
 
 const TicketGroups = {};
 
+const externalIdNamesToKeys = {
+	ifc_guids: ['IFC GUID', 'Ifc::IfcGUID', 'Element::IfcGUID'],
+	revit_ids: ['Element ID', 'Element ID::Value'],
+};
+
 const getIdToMeshesMapping = async (teamspace, model, revId) => {
 	const fileData = await getFile(teamspace, `${model}.stash.json_mpc`, `${UUIDToString(revId)}/idToMeshes.json`);
 	return JSON.parse(fileData);
 };
 
-const getObjectArrayFromRules = async (teamspace, project, model, revId, rules) => {
+const getExteralidNameFromMetadata = (metadata) => {
+	let externalIdName;
+	Object.keys(externalIdNamesToKeys).forEach((name) => {
+		if (externalIdNamesToKeys[name].some((m) => m === metadata[0].metadata[0].key)) {
```

Comment:
what if we only have partial match? i.e. we have 10 3drepo ids, 5 of which have IFC guids, 5 of which don't.

This is possible from Navisworks upload, where it is a federation of multiple files, and the files could be of different file type. E.g. if i upload a nwd that has a DGN and an IFC file. With this assumption,  if i create a manually selected group that contains DGN and IFC objects, i'm going to lose all the DGN records from the group?

I think we need to ensure there's a 100% match for a particular type of ID before we declare a match.

Reply:
`metadata[0].metadata[0]` is weird.

What if I am using an IFC file that has both revit and IFC guids? can you guarantee metadata[0] will always be the IFC GUID?

Or am I missing something?